src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -2819,7 +2916,7 @@ ss::future<bool> tx_gateway_frontend::try_create_tx_topic() {
     cluster::topic_configuration topic{
       model::kafka_internal_namespace,
       model::tx_manager_topic,
-      1,
+      config::shard_local_cfg().transaction_coordinator_partitions(),
```

Comment:
Please check the flag first, if it's off we should use `1` instead of `transaction_coordinator_partitions` because it isn't certain that the cluster will be upgraded.